lib/Controller/PersonApiController.php

```diff
@@ -0,0 +1,117 @@
+<?php
+/**
+ * @copyright Copyright (c) 2018-2020 Matias De lellis <mati86dl@gmail.com>
+ *
+ * @author Matias De lellis <mati86dl@gmail.com>
+ *
+ * @license GNU AGPL version 3 or any later version
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+namespace OCA\FaceRecognition\Controller;
+
+use OCA\FaceRecognition\Db\Face;
+use OCA\FaceRecognition\Db\FaceMapper;
+use OCA\FaceRecognition\Db\Image;
+use OCA\FaceRecognition\Db\ImageMapper;
+use OCA\FaceRecognition\Db\Person;
+use OCA\FaceRecognition\Db\PersonMapper;
+use OCA\FaceRecognition\Service\SettingsService;
+use OCA\FaceRecognition\Service\UrlService;
+use OCP\AppFramework\Http;
+use OCP\AppFramework\Http\DataResponse;
+use OCP\AppFramework\OCSController;
+use OCP\IRequest;
+
+class PersonApiController extends OCSController {
+
+	/** @var FaceMapper */
+	private $faceMapper;
+
+	/** @var ImageMapper */
+	private $imageMapper;
+
+	/** @var PersonMapper */
+	private $personMapper;
+
+	/** @var SettingsService */
+	private $settingsService;
+
+	/** @var UrlService */
+	private $urlService;
+
+	/** @var string */
+	private $userId;
+
+	public function __construct($AppName,
+	                            IRequest        $request,
+	                            FaceMapper      $faceMapper,
+	                            ImageMapper     $imageMapper,
+	                            PersonMapper    $personmapper,
+	                            SettingsService $settingsService,
+	                            UrlService      $urlService,
+	                            $UserId)
+	{
+		parent::__construct($AppName, $request);
+
+		$this->faceMapper      = $faceMapper;
+		$this->imageMapper     = $imageMapper;
+		$this->personMapper    = $personmapper;
+		$this->settingsService = $settingsService;
+		$this->urlService      = $urlService;
+		$this->userId          = $UserId;
+	}
+
+	/**
+	 * Get all persons with faces and file asociated
+	 *
+	 * @NoAdminRequired
+	 */
+	public function getPersons() {
```

Comment:
You are doing a simple query, which returns all the persons, all the faces and all the images.. 🙈 

This can be simple to consume in an application, but very very hard on the server if you have 50,000 photos, 😞

Reply:
Can you do something similar to the main application?

1. Get the "persons"...
2. When necessary, the photos of a person.
3. When necessary,  the faces/persons in a photo.

Reply:
> You are doing a simple query, which returns all the persons, all the faces and all the images

I was following the old API hoping that it'll be accepted more easily :wink:. If it's ok to change it, I would also prefer splitting it into multiple APIs

> but very very hard on the server if you have 50,000 photos

Hard to say. Currently there's not much data per row, so I believe the performance impact shouldn't be that much. But nevertheless, I agree that this is not a good approach